gnomad/assessment/sanity_checks.py

```diff
@@ -103,90 +117,832 @@ def make_filters_sanity_check_expr(
+            ),
+        }
+    return field_check_expr
 
 
 def compare_row_counts(ht1: hl.Table, ht2: hl.Table) -> bool:
     """
     Check if the row counts in two Tables are the same.
 
-    :param ht1: First Table to be checked
-    :param ht2: Second Table to be checked
-    :return: Whether the row counts are the same
+    :param ht1: First Table to be checked.
+    :param ht2: Second Table to be checked.
+    :return: Whether the row counts are the same.
     """
     r_count1 = ht1.count()
     r_count2 = ht2.count()
     logger.info("%d rows in left table; %d rows in right table", r_count1, r_count2)
     return r_count1 == r_count2
+
+
+def summarize_variant_filters(
+    t: Union[hl.MatrixTable, hl.Table],
+    variant_filter_field: str = "RF",
+    problematic_regions: List[str] = ["lcr", "segdup", "nonpar"],
+    single_filter_count: bool = False,
+    monoallelic_expr: Optional[hl.expr.BooleanExpression] = None,
+    large_n_rows: int = 50,
+    large_n_cols: int = 140,
+) -> None:
+    """
+    Summarize variants filtered under various conditions in input MatrixTable or Table.
+
+    Summarize counts for:
+        - Total number of variants
+        - Fraction of variants removed due to:
+            - Any filter
+            - Inbreeding coefficient filter in combination with any other filter
+            - AC0 filter in combination with any other filter
+            - `variant_filter_field` filtering in combination with any other filter in combination with any other filter
+            - Only inbreeding coefficient filter
+            - Only AC0 filter
+            - Only `variant_filter_field` filtering
+
+    :param t: Input MatrixTable or Table to be checked.
+    :param variant_filter_field: String of variant filtration used in the filters annotation on `ht` (e.g. RF, VQSR, AS_VQSR). Default is "RF".
+    :param problematic_regions: List of regions considered problematic to run filter check in. Default is ["lcr", "segdup", "nonpar"].
+    :param single_filter_count: If True, explode the Table's filter column and give a supplement total count of each filter. Default is False.
+    :param monoallelic_expr: Optional boolean expression of monoallelic status that logs how many monoallelic sites are in the Table.
+    :param large_n_rows: Number of rows to show when showing percentages of filtered variants grouped by multiple conditions. Default is 50.
+    :param large_n_cols: Number of columns to show when showing percentages of filtered variants grouped by multiple conditions. Default is 140.
+    :return: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    filters = t.aggregate(hl.agg.counter(t.filters))
+    logger.info("Variant filter counts: %s", filters)
+
+    if single_filter_count:
+        ex_t = t.explode(t.filters)
+        filters = ex_t.aggregate(hl.agg.counter(ex_t.filters))
+        logger.info("Exploded variant filter counts: %s", filters)
+
+    if monoallelic_expr is not None:
```

Comment:
hmm good question. The gnomAD dataset doesn't include monoallelic as a filter, it is a separate annotation but this would still be doable. @jkgoodrich , what do you think?

Reply:
I think we can leave as is since we decided not to actually have this as a filter and we can revisit if we decide to change that for future callsets